integr8ly/application-monitoring-operator
# INTLY-4029 Create Alertmanager & Prometheus serviceaccounts if they d…

…on't exist

This change is intended for clusters already running an older version of AMO (e.g. Integreatly/RHMI) & getting the config up to date wrt newly named serviceaccounts for Alertmanager & Prometheus.

Steps to verify:

- Provision integreatly from 1.5.1
- Scale down application-monitoring-operator in cluster
- ~Patch the AlertManager & Prometheus CR's to use the new serviceaccount names~ I'll add this to the operator as it owns both those CR's.
- Run this branch locally against the cluster `operator-sdk up local --namespace=middleware-monitoring`
- Verify the below log output showing serviceaccounts were created

```
{"level":"info","ts":1572881345.450652,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:199","msg":"Error retrieving serviceaccount: alertmanager-service-account : ServiceAccount \"alertmanager-service-account\" not found"}
{"level":"info","ts":1572881345.450696,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:201","msg":"Creating serviceaccount: alertmanager-service-account"}
{"level":"info","ts":1572881346.476511,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:216","msg":"Service account: alertmanager-service-account exists, annotations: map[]"}

{"level":"info","ts":1572881376.4809442,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:214","msg":"Key: serviceaccounts.openshift.io/oauth-redirectreference.primary exists. Val: {\"kind\":\"OAuthRedirectReference\",\"apiVersion\":\"v1\",\"reference\":{\"kind\":\"Route\",\"name\":\"alertmanager-route\"}} Do nothing."}
{"level":"info","ts":1572881376.480972,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:199","msg":"Error retrieving serviceaccount: prometheus-service-account : ServiceAccount \"prometheus-service-account\" not found"}
{"level":"info","ts":1572881376.481091,"logger":"controller_applicationmonitoring","caller":"applicationmonitoring/applicationmonitoring_controller.go:201","msg":"Creating serviceaccount: prometheus-service-account"}
```
